src/NServiceBus.AcceptanceTests/Pipeline/When_replacing_behavior.cs

```diff
@@ -0,0 +1,97 @@
+﻿namespace NServiceBus.AcceptanceTests.Pipeline
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.AcceptanceTesting;
+    using EndpointTemplates;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+
+    public class When_replacing_behavior : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_invoke_replacement_in_pipeline()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointWithReplacement>(e => e
+                    .When(s => s.SendLocal<Message>(m => { })))
+                .Done(c => c.MessageHandled)
+                .Run();
+
+            Assert.IsFalse(context.OriginalBehaviorInvoked);
+            Assert.IsTrue(context.ReplacementBehaviorInvoked);
+        }
+
+        class Context : ScenarioContext
+        {
+            public bool OriginalBehaviorInvoked { get; set; }
+            public bool ReplacementBehaviorInvoked { get; set; }
+
+            public bool MessageHandled { get; set; }
+        }
+
+        class OriginalBehavior : Behavior<ITransportReceiveContext>
+        {
+            Context testContext;
+
+            public OriginalBehavior(Context testContext)
+            {
+                this.testContext = testContext;
+            }
+
+            public override Task Invoke(ITransportReceiveContext context, Func<Task> next)
+            {
+                testContext.OriginalBehaviorInvoked = true;
+                return next();
+            }
+        }
+
+        class ReplacementBehavior : Behavior<ITransportReceiveContext>
+        {
+            Context testContext;
+
+            public ReplacementBehavior(Context testContext)
+            {
+                this.testContext = testContext;
+            }
+
+            public override Task Invoke(ITransportReceiveContext context, Func<Task> next)
+            {
+                testContext.ReplacementBehaviorInvoked = true;
+                return next();
+            }
+        }
+
+        class EndpointWithReplacement : EndpointConfigurationBuilder
+        {
+            public EndpointWithReplacement()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.Pipeline.Replace("demoBehavior", typeof(ReplacementBehavior));
+                    c.Pipeline.Register("demoBehavior", typeof(OriginalBehavior), "test behavior replacement");
```

Comment:
I did it intentionally to make sure replace works out of order (because of feature activation which also may run out of order).  Are you ok if I add a comment about this instead?